sphinx_gallery/tests/test_gen_rst.py

```diff
@@ -260,6 +261,22 @@ def test_save_figures():
 
     shutil.rmtree(examples_dir)
 
+
+def test_zip_notebooks():
+    """Test generated zipfiles are not corrupt"""
+    gallery_conf = build_test_configuration(examples_dir='examples')
+    examples = [fname
+                for fname in sorted(os.listdir(gallery_conf['examples_dir']))
+                if fname.endswith('.py')]
+    sg.python_zip(examples, gallery_conf['examples_dir'],
+                  gallery_conf['gallery_dir'])
+    zipfilename = os.path.join(gallery_conf['gallery_dir'], "_python.zip")
```

Comment:
I just realized: are the tests creating files on the disk? If so, they should remove them after, in a try/finally setting.

Ideally, these files should be created in a temporary folder, as sometimes the build folder is not writable.